lib/Infer/InstSynthesis.cpp

```
@@ -0,0 +1,1264 @@
+              continue;
+            llvm::outs() << "setting input " << Input.first->Name
+                         << " to " << Input.second->Val << "\n";
+          }
+        }
+        if (J > 0) {
+          // Starting with the second concrete input set,
+          // copy component input variables for query separation
+          for (auto const &E : LocInstMap) {
+            if (E.first.find(COMP_INPUT_PREFIX) != std::string::npos) {
+              auto In = E.second.second;
+              std::string Name = E.first + LOC_SEP + std::to_string(Refinements);
+              InputMap[In] = IC.createVar(In->Width, Name);
+            }
+          }
+        }
+        Inst *Copy = getInstCopy(WiringQuery, IC, InputMap);
+        Query = IC.getInst(Inst::And, 1, {Query, Copy});
+      }
+      // Solve the synthesis constraint.
+      // Each solution corresponds to a syntactically distinct and well-formed
+      // straight-line program obtained by composition of given components
+      std::vector<Inst *> ModelInsts;
+      std::vector<llvm::APInt> ModelVals;
+      InstMapping Mapping(Query, IC.getConst(APInt(1, true)));
+      // Negate the query to get a SAT model.
+      // Don't use original BPCs/PCs, they are useless
+      std::string QueryStr = BuildQuery({}, LoopPCs, Mapping,
+                                        &ModelInsts, /*Negate=*/true);
+      bool IsSat;
+      EC = SMTSolver->isSatisfiable(QueryStr, IsSat, ModelInsts.size(),
+                                    &ModelVals, Timeout);
+      if (EC)
+        return EC;
+
+      // No valid wiring exists for the target comp number
+      if (!IsSat)
+        break;
+
+      std::vector<std::pair<LocInst, LocInst>> Wiring;
+      Inst *Cand = createInstFromModel(std::make_pair(ModelInsts, ModelVals),
+                                       Wiring, IC);
+      if (!Cand) {
+        llvm::errs() << "synthesis bug: creating inst from a model failed\n";
```

If this truly should never happen, I'd prefer to crash souper when it does, instead of printing something.